hashing.go

```diff
@@ -0,0 +1,96 @@
+package redis
+
+import "sort"
+
+// hashRing is the implementation of a consistent hashing distribution of string
+// keys to server addresses.
+type hashRing []hashNode
+
+type hashNode struct {
+	addr string
+	hash uint64
+}
+
+const (
+	hashRingReplication = 40
```

Comment:
It gave good results, that's the value another hash ring package I found was using as well.